coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -174,10 +174,15 @@ class MultiPartitionPlanner(partitionLocationProvider: PartitionLocationProvider
       logger.debug(s"partitionInfo=$p; queryParams=$queryParams")
       if (p.partitionName.equals(localPartitionName))
         localPartitionPlanner.materialize(lp.copy(startMs = p.timeRange.startMs, endMs = p.timeRange.endMs), qContext)
-      else
-        createMetadataRemoteExec(qContext, queryParams, p,
-          Map("filter" -> lp.filters.map{f => f.column + f.filter.operatorString + f.filter.valuesStrings.head}.
-            mkString(","), "labels" -> lp.labelNames.mkString(",")))
+      else {
+        val filters = if (queryParams.remoteQueryPath.get.contains("""/v2/label/""")) lp.filters.map{ f =>
```

Comment:
Can you please avoid duplicate code if possible.

Reply:
Done